src/System.Private.CoreLib/shared/System/AggregateException.cs

```diff
@@ -257,8 +257,7 @@ private AggregateException(string message, IList<ExceptionDispatchInfo> innerExc
                 throw new ArgumentNullException(nameof(info));
             }
 
-            Exception[]? innerExceptions = info.GetValue("InnerExceptions", typeof(Exception[])) as Exception[];
-            if (innerExceptions == null)
+            if (!(info.GetValue("InnerExceptions", typeof(Exception[])) is Exception[] innerExceptions))
```

Comment:
To be honest, I don't see how this is more readable.

Reply:
Reverted.